docs/contributing.md

```diff
@@ -0,0 +1,121 @@
+  rustup component add clippy                 # Install the linter.
+  ```
+
+- **Node Version Manager**  
+  In order to build the web and desktop applications you will need [node](https://nodejs.org) and 
+[npm](https://www.npmjs.com). Please note that the versions available in your system package manager
+will probably crash while building native-extensions. The only known stable solution is to use the 
+[Node Version Manager](https://github.com/nvm-sh/nvm). Please note that installing it from any 
+package manager is officially not supported and can cause issues. Follow the 
+[official guide](https://github.com/nvm-sh/nvm#installing-and-updating) to install it:
+
+  ```bash
+  curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.2/install.sh | bash
+  ```
+
+<br/>
+<br/>
+<br/>
+
+## Working with sources
+As this is a multi-part project with many complex dependencies, it was equipped with a build script
+which both validates your working environment as well as takes care of providing most suitable 
+compilation flags for a particular development stage. In order to learn more about the commands and 
+available options, simply run `./run` and read the manual. The most common options are presented 
+below:
+
+- **Interactive builds**  
+  Run `./run watch` to start a local web-server and a source-file watch utility which will build the 
+  project on every change. Open `http://localhost:8080` to run the application, or 
+  `http://localhost:8080/debug` to see example demo scenes. Please remember to disable the cache in 
+  your browser during development! By default, the script disables heavyweight optimizations to 
+  provide interactive development experience. The scripts are thin wrappers for 
+  [wasm-pack](https://github.com/rustwasm/wasm-pack) and accept the same 
+  [command line arguments](https://rustwasm.github.io/wasm-pack/book/commands/build.html).
+
+- **Production builds**  
+  In order to compile in a production mode (enable all optimizations, strip WASM debug symbols, 
+  minimize the output binaries), run `./run build` or `./run dist`. The latter creates also packages
+  and installers for all supported platforms.
+
+
+## Testing, Linting, and Validation
+After changing the code it's always a good idea to lint and test the code. We have prepared several 
+scripts which maximally automate the process:
+
+- **Size Validation**
+  Use `run check-size` to check if the size of the final binary did not grew too much in comparison
+  to the previous release. Watching the resulting binary size is one of the most important 
+  responsibility of each contributor in order to keep the project small and suitable for web-based
+  usage.
+  
+- **Testing**
+  Use `run test` run both unit and web-based test.
+  
+<br/>
+
+### Testing
+The sources use both unit tests and web test, which are run in a browser and
+produce visual results. To run them, use the `scripts/test.sh` script and follow
```

Comment:
Here and further — scripts -> build
Also, it is really unclear how different is this from `run test`.